internal/webhook/external/handlers/refresh_warehouse.go

```diff
@@ -0,0 +1,108 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/api"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/indexer"
+	"github.com/akuity/kargo/internal/logging"
+	"github.com/akuity/kargo/internal/webhook/external/providers"
+)
+
+// NewRefreshWarehouseWebhook handles push events for the designated provider.
+// After the provider has been resolved and the request has been authenticated,
+// the kubeclient is queried for all warehouses that contain a subscription
+// to the repo in question. Those warehouses are then patched with a special
+// annotation that signals down stream logic to refresh the warehouse.
+func NewRefreshWarehouseWebhook(p providers.Provider, c client.Client) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger := logging.LoggerFromContext(r.Context())
+		logger.Debug("authenticating request")
+		w.Header().Set("Content-type", "application/json")
```

Comment:
It's fine to do this unconditionally here because we _are_ going to respond with JSON no matter what, but I'd propose that your `xhttp.Error()` function should probably set it as well to eliminate the possibility of calling `xhttp.Error()` (in some other context) after having _mistakenly forgotten_ to set the content type.